File: RectangleCpp/Font.cpp
#include "Font.h"

#pragma region private
void Font::throwExceptionIfIllegalSize(int size) {
	if (size < 0)
		throw new std::invalid_argument("Font size cannot be < 0");
}
#pragma endregion

#pragma region public
Font::Font(std::string name, Color color, int size) {
	throwExceptionIfIllegalSize(size);
	this->name = name;
	this->color = color;
	this->size = size;
}

void Font::setName(std::string name) {
	this->name = name;
}

void Font::setColor(Color color) {
	this->color = color;
}

void Font::setSize(int size) {
	throwExceptionIfIllegalSize(size);
	this->size = size;
}

std::string Font::getName() {
	return name;
}

Color Font::getColor() {
	return color;
}

int Font::getSize() {
	return size;
}
#pragma endregion public

File: Rectangle/src/main/java/me/divium/rectangle/Font.java
package me.divium.rectangle;

public class Font {
    protected String name;
    protected Color color;
    protected int size;

    protected void throwExceptionIfIllegalSize(int size) {
        if (size < 0)
            throw new IllegalArgumentException("Font size cannot be < 0");
    }

    public Font(String name, Color color, int size) {
        throwExceptionIfIllegalSize(size);
        this.name = name;
        this.color = color;
        this.size = size;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setColor(Color color) {
        this.color = color;
    }

    public void setSize(int size) {
        throwExceptionIfIllegalSize(size);
        this.size = size;
    }

    public String getName() {
        return name;
    }

    public Color getColor() {
        return color;
    }

    public int getSize() {
        return size;
    }
}
